composer/utils/object_store/mlflow_object_store.py

```diff
@@ -0,0 +1,392 @@
+
+        tracking_uri = os.getenv('MLFLOW_TRACKING_URI', MLFLOW_DATABRICKS_TRACKING_URI)
+        if tracking_uri != MLFLOW_DATABRICKS_TRACKING_URI:
+            raise ValueError(
+                'MLFlowObjectStore currently only supports Databricks-hosted MLFlow tracking. '
+                f'Environment variable `MLFLOW_TRACKING_URI` is set to a non-Databricks URI {tracking_uri}. '
+                f'Please unset it or set the value to `{MLFLOW_DATABRICKS_TRACKING_URI}`.')
+
+        # Use the Databricks WorkspaceClient to check that credentials are set up correctly.
+        try:
+            WorkspaceClient()
+        except Exception as e:
+            raise ValueError(
+                f'Databricks SDK credentials not correctly setup. '
+                'Visit https://databricks-sdk-py.readthedocs.io/en/latest/authentication.html#databricks-native-authentication '
+                'to identify different ways to setup credentials.') from e
+
+        self._mlflow_client = MlflowClient(MLFLOW_DATABRICKS_TRACKING_URI)
+        mlflow.environment_variables.MLFLOW_MULTIPART_UPLOAD_CHUNK_SIZE.set(multipart_upload_chunk_size)
+
+        experiment_id, run_id, _ = MLFlowObjectStore.parse_dbfs_path(path)
+        if experiment_id == PLACEHOLDER_EXPERIMENT_ID:
+            experiment_id = None
+        if run_id == PLACEHOLDER_RUN_ID:
+            run_id = None
+
+        # Construct the `experiment_id` and `run_id` depending on whether format placeholders were provided.
+        if not dist.is_initialized() or dist.get_global_rank() == 0:
+            experiment_id, run_id = self._init_run_info(experiment_id, run_id)
+
+        if dist.is_initialized():
+            mlflow_info = [experiment_id, run_id]
+            dist.broadcast_object_list(mlflow_info, src=0)
+            experiment_id, run_id = mlflow_info
+
+        self.experiment_id = experiment_id
+        self.run_id = run_id
+
+    def _init_run_info(self, experiment_id: Optional[str], run_id: Optional[str]) -> Tuple[str, str]:
+        """Returns the experiment ID and run ID for the MLFlow run backing this object store.
+
+        In a distributed setting, this should only be called on the rank 0 process.
+        """
+        import mlflow
+
+        if experiment_id is None:
+            if run_id is not None:
+                raise ValueError('A `run_id` cannot be provided without a valid `experiment_id`.')
+
+            active_run = mlflow.active_run()
+            if active_run is not None:
+                experiment_id = active_run.info.experiment_id
+                run_id = active_run.info.run_id
+                log.debug(f'MLFlowObjectStore using active MLFlow run (run_id={run_id})')
+            else:
+                # If no active run exists, create a new run for the default experiment.
+                experiment_name = os.getenv(mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.name,
```

Comment:
im guessing there is some shared logic in the init with the mlflow logger init. anything we can pull out into a helper?

Reply:
It's a little awkward since the inits take inherently different inputs, so the logic is actually pretty different. The logger accepts init from run/experiment name and then may or may not create the experiment/run depending on whether ID environment variables are set, while the object store depends entirely on the URI with the experiment/run IDs.